rmm-config.cmake.in

```diff
@@ -7,6 +7,39 @@ include(CMakeFindDependencyMacro)
 find_dependency(CUDAToolkit)
 find_dependency(spdlog 1.7.0)
 
+# find_dependency(Thrust 1.10.0)
+# thrust_create_target(rmm::Thrust FROM_OPTIONS)
+
+# Workaround for possibly missing Thrust
+# We essentially do find_dependency ourselves, so that
+# we can customize what happens if we don't find Thrust
+if(rmm_FIND_QUIETLY)
+  set(cmake_fd_quiet_arg QUIET)
+endif()
+
+get_property(cmake_fd_alreadyTransitive GLOBAL PROPERTY
+  _CMAKE_Thrust_TRANSITIVE_DEPENDENCY
+  )
+
+find_package(Thrust 1.10.0 ${cmake_fd_quiet_arg})
+
+if(NOT DEFINED cmake_fd_alreadyTransitive OR cmake_fd_alreadyTransitive)
+  set_property(GLOBAL PROPERTY _CMAKE_Thrust_TRANSITIVE_DEPENDENCY TRUE)
+endif()
+
+set(cmake_fd_quiet_arg)
+
+if (Thrust_FOUND)
+  thrust_create_target(rmm::Thrust FROM_OPTIONS)
+else()
+  message(WARNING "RMM could not find a Thrust library >= 1.10.0. Continuing anyway, which likely means that the Thrust library from your CUDA Toolkit will be used.")
```

Comment:
Can we tweak this message to give more information about what thrust version was found and where it was found? i.e.
```
  message(WARNING "RMM could not find a Thrust library >= 1.10.0. Using Thrust version ${Thrust_VERSION} found at ${Thrust_PATH}.")
```

Reply:
Not easily, since we haven't found a thrust at this point.

Reply:
I also don't really like the hard-coded "1.10.0" in the message. Can you parameterize that based on the specified dependency version?

Reply:
So a few ideas here:

1. We can keep the error message as is but include at least the cuda toolkit version since we already had `find_dependency(CUDAToolkit)`. This would help for debugging purposes if people are reporting broken builds due to RMM downstream.
2. We could explicitly use the cudatoolkit paths to find the `include/thrust` folder and manually create the target based on that instead of creating an empty Thrust target.
3. We could write / use a `FindThrust` cmake module to search through the standard system locations as well as CUDA install locations.

Additionally, if we went with option 2 or 3, we could capture the version by parsing `include/thrust/version.h` with a basic regex to grab the version number and then present a warning that version x.y.z is being used while we only test >= 1.10.0.

I'm strongly against option 3 and lean towards option 2.

Reply:
Given that @harrism says that RMM should work with thrust 1.9.x, I'm inclined to support it properly then. Essentially all of 1., 2., 3. are kinda similar in that they're trying to find thrust, anyway, and ideally extract the version. If that code exists anyway, it might as well be done inside of `FindThrust.cmake` to make the rmm-config not even uglier than it already is. In fact, things will look much cleaner, I'm pretty sure.